modules/swagger-codegen/src/main/resources/python/api_client.mustache

```diff
@@ -625,3 +640,124 @@ class ApiClient(object):
             if klass_name:
                 instance = self.__deserialize(data, klass_name)
         return instance
+ 
+{{#httpsig}}    
+    def get_sha256_digest(self, data):
+        """
+        :param data: Data set by User
+        :return: instance of digest object 
+        """
+        digest = SHA256.new()
+        digest.update(data)
+        
+        return digest
+    
+    def get_rsasig_b64encode(self, private_key_path, digest):
+        """
+        :param private_key_path : abs path to private key .pem file.
+        :param digest: digest
+        :return: instance of digest object 
+        """
+        
+        key = open(private_key_path, "r").read()
+        rsakey = RSA.importKey(key)
+        signer = PKCS1_v1_5.new(rsakey)
+        sign = signer.sign(digest)
+        
+        return b64encode(sign)
+    
+    def prepare_str_to_sign(self, req_tgt, hdrs):
+        """
+        :param req_tgt : Request Target as stored in http header.
+        :param hdrs: HTTP Headers to be signed.
+        :return: instance of digest object 
+        """
+        ss = ""
+        ss = ss + "(request-target): " + req_tgt.lower() + "\n"
+    
+        length = len(hdrs.items())
+    
+        i = 0
+        for key, value in hdrs.items():
+            ss = ss + key.lower() + ": " + value
+            if i < length-1:
+                ss = ss + "\n"
+            i += 1
+        
+        return ss
```

Comment:
This whole function is unpythonic. Use string formatting and '\n'.join(...). String concatenation is both ugly and inefficient in Python.

Reply:
Ok. Will change.